pkg/autodetect/main.go

```diff
@@ -117,10 +117,10 @@ func (v AutoscalingVersion) String() string {
 	return "unknown"
 }
 
-func toAutoscalingVersion(version string) AutoscalingVersion {
+func ToAutoScalingVersion(version string) AutoscalingVersion {
```

Comment:
any reason why the function is public now?

Reply:
So it can be accessed by the test, as the test is in a different package

Reply:
It's a bit unfortunate/unnecessary to have the test in a different package in this case. But it was done like that from before so I will approve the PR.